update.ps1

```diff
@@ -0,0 +1,98 @@
+$script:rancherDesktopInstallerHash = "92108CBBD8C98F99B00A608D8F7D21E12FAECA76F16890585EF212CC5BF1C779"
+
+#endregion
+
+#region functions
+
+function RenameBinariesFunction
+{
+    Write-Host "Renaming the Rancher Desktop binaries..." -ForegroundColor Blue
+    Rename-Item -Path "$script:windowsBinariesPath\docker.exe" -NewName dockerw.exe
+    Rename-Item -Path "$script:windowsBinariesPath\docker-compose.exe" -NewName dockerw-compose.exe
+    Copy-Item "$script:windowsBinariesPath\nerdctl.exe" "$script:windowsBinariesPath\docker.exe" -Force
+
+    Rename-Item -Path "$script:linuxBinariesPath\docker" -NewName docker.old
+    Copy-Item "$script:linuxBinariesPath\nerdctl" "$script:linuxBinariesPath\docker" -Force
+    Write-Host "Renaming done." -ForegroundColor Green
+}
+
+function UpdateRancherDesktop
+{
+
+    Write-Host "Updating Rancher Desktop to version $script:rancherDesktopTargetVersion..." -ForegroundColor Blue
+    
+    if(!(Test-Path -Path $script:rancherDesktopExe))
+    {
+        Write-Host "Rancher Desktop was not found in the system." -ForegroundColor Red
+        exit 1
+    }
+    elseif ((Get-FileHash -Algorithm SHA256 "$script:rancherDesktopExe").Hash -ne "$script:rancherDesktopExeHash")
+    {
+        Write-Host "Wrong Rancher Desktop version detected. Please make sure that the version installed is: $script:rancherDesktopVersion" -ForegroundColor Red
+        exit 2      
+    }
+    else # All update preconditions passed
+    {
+        # Create backup of dockerd
+        Copy-Item "$script:windowsBinariesPath\dockerd.exe" "C:\" -Recurse -Force
+
+        $BinariesRenamed = Test-Path -Path "$script:linuxBinariesPath\docker.old"
+        
+        if(!(Test-Path -Path "$script:rancherDesktopInstallerName.exe") -or (Get-FileHash -Algorithm SHA256 "$script:rancherDesktopInstallerName.exe").Hash -ne "$script:rancherDesktopInstallerHash")
+        {
+            Invoke-WebRequest $script:rancherDesktopUrl -OutFile "$script:rancherDesktopInstallerName.exe"
+            if((Get-FileHash -Algorithm SHA256 "$script:rancherDesktopInstallerName.exe").Hash -ne "$script:rancherDesktopInstallerHash")
+            {
+                Write-Host "Checksum validation of Rancher Desktop installer failed." -ForegroundColor Red
+                exit 3
+            }
+        }
+        
+        Invoke-Expression ".\$script:rancherDesktopInstallerName.exe"
+
+        $setupId = (Get-Process $script:rancherDesktopInstallerName).id 2> $null
+
+        Wait-Process -Id $setupId
+
+        Write-Host "Rancher Desktop successfully updated to version $script:rancherDesktopTargetVersion." -ForegroundColor Green
+
+        # Restore and remove backup of dockerd
+        Stop-Service -Name "docker"
+        Copy-Item "C:\dockerd.exe" $script:windowsBinariesPath  -Recurse -Force
+        Remove-Item "C:\dockerd.exe"
+        Start-Service -Name "docker"
```

Comment:
BTW, this and the above assume that the user installed Windows containers support.